portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2938,6 +2962,137 @@ private DLVersionNumberIncrease _computeDLVersionNumberIncrease(
 			previousDLFileVersion, nextDLFileVersion);
 	}
 
+	private void _expireFileEntriesByCompanyId(
+			Date expirationDate, Map<String, Serializable> workflowContext,
+			ServiceContext serviceContext, Long companyId)
+		throws PortalException {
+
+		long userId = _getActiveCompanyAdminUserId(companyId);
+
+		List<DLFileEntry> fileEntries =
+			_getFileEntriesByCompanyIdAndExpirationDate(
+				companyId, expirationDate);
+
+		for (DLFileEntry fileEntry : fileEntries) {
+			DLFileVersion latestFileVersion =
+				dlFileVersionLocalService.fetchLatestFileVersion(
+					fileEntry.getFileEntryId(), false);
+
+			if (latestFileVersion.isExpired()) {
+				continue;
+			}
+
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					StringBundler.concat(
+						"Expiring file entry ", fileEntry.getFileEntryId(),
+						" with expiration date ",
+						fileEntry.getExpirationDate()));
+			}
+
+			updateStatus(
+				userId, fileEntry, latestFileVersion,
+				WorkflowConstants.STATUS_EXPIRED, serviceContext,
+				workflowContext);
+
+			_notifySubscribers(
+				userId, latestFileVersion, _buildEntryURL(latestFileVersion),
+				"expire", new ServiceContext());
+
+			_notifyOwner(
+				userId, latestFileVersion, _buildEntryURL(latestFileVersion),
+				"expire", new ServiceContext());
+		}
+	}
+
+	private long _getActiveCompanyAdminUserId(long companyId)
+		throws PortalException {
+
+		Role role = roleLocalService.getRole(
+			companyId, RoleConstants.ADMINISTRATOR);
+
+		long[] userIds = userLocalService.getRoleUserIds(role.getRoleId());
+
+		User user = null;
+
+		if (!ArrayUtil.isEmpty(userIds)) {
+			for (long userId : userIds) {
+				user = userLocalService.fetchUser(userId);
+
+				if ((user != null) && user.isActive()) {
+					return userId;
+				}
+			}
+		}
+
+		List<Group> groups = groupLocalService.getRoleGroups(role.getRoleId());
+
+		for (Group group : groups) {
+			if (group.isOrganization()) {
+				userIds = organizationLocalService.getUserPrimaryKeys(
+					group.getClassPK());
+
+				if (!ArrayUtil.isEmpty(userIds)) {
+					for (long userId : userIds) {
+						user = userLocalService.fetchUser(userId);
+
+						if ((user != null) && user.isActive()) {
+							return userId;
+						}
+					}
+				}
+			}
+			else if (group.isRegularSite()) {
+				userIds = groupLocalService.getUserPrimaryKeys(
+					group.getGroupId());
+
+				if (!ArrayUtil.isEmpty(userIds)) {
+					for (long userId : userIds) {
+						user = userLocalService.fetchUser(userId);
+
+						if ((user != null) && user.isActive()) {
+							return userId;
+						}
+					}
+				}
+			}
+			else if (group.isUserGroup()) {
```

Comment:
Can you add an additional case for Asset Libraries?

Reply:
thanks!

I totally forgot about asset libraries :D